azure_data_cosmos_engine/src/query/producer/mod.rs

```
@@ -64,33 +64,50 @@ pub fn create_partition_state(
 
 pub fn create_query_chunk_states(
     query_chunks: &Vec<QueryChunk>,
+    pk_paths: Vec<String>,
 ) -> Vec<QueryChunkState> {
     let mut chunk_states = Vec::with_capacity(query_chunks.len());
 
     for i in 0..query_chunks.len() {
-        let query = create_query_chunk_query(&query_chunks[i].items);
+        let query = create_query_chunk_query(&query_chunks[i].items, &pk_paths);
         // For QueryChunkState, we will use the index as the indentifier as opposed to pkrange ID.
-        chunk_states.push(QueryChunkState::new(
+        let chunk = QueryChunkState::new(
             i,
             query_chunks[i].pk_range_id.clone(),
             query,
-        ));
+        );
+        tracing::debug!("created query chunk state: {:?}", chunk);
+        chunk_states.push(chunk);
     }
     chunk_states
 }
 
-fn create_query_chunk_query(query_chunk_items: &Vec<QueryChunkItem>) -> String {
+fn create_query_chunk_query(query_chunk_items: &Vec<QueryChunkItem>, pk_paths: &Vec<String>) -> String {
     if query_chunk_items.is_empty() {
         return "SELECT * FROM c WHERE 1 = 0".to_string();
     }
-
-    let conditions = query_chunk_items
-        .iter()
-        .map(|item| format!("( c.id = '{}' )", item.id))
-        .collect::<Vec<_>>()
-        .join(" OR ");
-
-    format!("SELECT * FROM c WHERE ( {conditions} )")
+    
+    if pk_paths.len() == 1 {
+        // strip the leading "/" to get just the partition key property name
+        let pk_path = pk_paths[0].trim_start_matches('/');
+        
+        let conditions = query_chunk_items
+            .iter()
+            .map(|item| format!("(c.id = '{}' AND c.{} = '{}')", item.id, pk_path, item.partition_key_value))
+            .collect::<Vec<_>>()
+            .join(" OR ");
+
+        format!("SELECT * FROM c WHERE ( {conditions} )")
+    } else {
+        // here we could have logic for HPK later down the line - for now we just do queries with only ID values
```

We should return an error here if we we're sure we can return an accurate value. Better to tell the user "HPK is not supported" instead of returning potentially incorrect data.

Of course, because this is Rust, that means you'll have to make this return a `Result` and update callers, but I think it's worth it.